### PR TITLE
feat: add cloudformation for redis & vpc, add ssm env secrets

### PR DESCRIPTION
**Pull request by zachferland**
Adds cloudformation template to create vpc/subnets for functions, with with redis db in same for access. 
Adds ssm to easily manage secrets 

May make sense to make some of this optional in serverless config if possible, ie ssm, and deploying may cause more resource use expected. But this is a nice quick way for anyone to deploy to aws now. 

Not finished testing yet

### CODE REVIEW

**Comment by zachferland** [comment]
cc @valmack

**Comment by valmack** [comment]
@zachferland Cool! I'm not super well versed in cloud formation templates
but does the VPC creation need to live here?

It feels as if the person deploying should select an existing VPC and subnets from the cloud formation console. https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/parameters-section-structure.html#aws-specific-parameter-types

File: packages/server/serverless.yml [hunk 1]
```diff
@@ -9,12 +9,20 @@ plugins:
...
   stage: develop
-  region: us-west-2
-  # environment:
-  # SECRETS: ${self:custom.kmsSecrets.secrets.SECRETS}
-  # AWS_BUCKET_NAME: ${self:custom.awsBucketName.${self:provider.stage}}
+  region: us-east-2
+  environment:
+    REDIS_HOST:
+      "Fn::GetAtt": [ElasticCacheCluster, RedisEndpoint.Address]
+    VERIFICATION_ISSUER_DOMAIN: ${self:custom.secrets.VERIFICATION_ISSUER_DOMAIN}
```
Comment (oed) [inline packages/server/serverless.yml:18]: Where are these secrets being pulled from now?

**Review by oed — APPROVED** [review]
lgtm, but don't know much about cloudformation config :)

File: packages/server/serverless.yml (hunk 1 above)
Comment (zachferland) [reply packages/server/serverless.yml:18]: encrypted secrets object pulled below from ssm (aws secrets manager)

**Comment by zachferland** [comment]
I dont know much about cloudformation either :) mostly templates, tried some serverless plugins as well, but this was simplest 

re @valmack and future improvements:

yeah few parts here can be a bit more dev friendly and configurable
- should be able to serverless deploy with your own redis instance and vpc configuration, dont think i would pass any vars in cloudformation templates, but in serverless config you can pass VPC and SG, would be nice to have some simple logic that if you do pass a redist host yourself, then the cloudformation template is not deployed. Seems possible or at least we may want more than one serverless config file here, then just some npm commands for each and docs (likely more clear even)
- also would be nice to still allow env config file or aws SSM still, similar logic, if cant find secrets in ssm, then just fallback to env config file (behavior may be this already) 

But this doesnt feel to high priority right now, so will likely put in issue for later

**Comment by oed** [comment]
Any blockers here @zachferland ?

**Comment by zachferland** [comment]
All set for now, also includes support for passing verification urls rather than just looking up last 30 mins. Allows support for migrated links. Also namespaces challenges and has some aws fixes. 

May still change redis connection handling if time and pass mock data in tests form github/twitter if time, but can be in another pr.

**Comment by valmack** [comment]
@zachferland can we ensure this has tags before this goes in btw?

Ideally k: IdentityLink v: <env>

Or just the key on all the resources that take tags

**Comment by zachferland** [comment]
@valmack yeah i can tag  things, but also you can easily reference all resources from a single cloudformation deployment(ie delete them, look them up etc)

**Comment by zachferland** [comment]
Add stackTags (global tag) to aws provider which which will tag all taggable resources deployed by the cloudformation template here

**Pull request merged**